tstest/integration/testcontrol/testcontrol.go

```diff
@@ -67,6 +68,28 @@ func (s *Server) NumNodes() int {
 	return len(s.nodes)
 }
 
+// AddPingRequest sends the ping pr to nodeKeyDst. It reports whether it did so. That is,
+// it reports whether nodeKeyDst was connected.
+func (s *Server) AddPingRequest(nodeKeyDst tailcfg.NodeKey, pr *tailcfg.PingRequest) bool {
+	s.mu.Lock()
+	if s.pingReqsToAdd == nil {
+		s.pingReqsToAdd = map[tailcfg.NodeKey]*tailcfg.PingRequest{}
+	}
+	s.mu.Unlock()
+	// Now send the update to the channel
+	node := s.Node(nodeKeyDst)
```

Comment:
Ah, this is a little sad that we have to Unlock the mutex just to reacquire it right away.

Generally you don't want to drop + reacquire mutexes because invariants might've changed meanwhile that you have to double-check. In this case there aren't really anyway, but it's cleaner to do without.

I just pushed 58cc2cc9215843790d67d3a67865254d9512bd8d that you can use instead, to call `s.nodeLocked(nodeKeyDst)`, and then you can go back to using defer Unlock.

You'll need to rebase this atop head.